### ISSUE TITLE
I want to reproduce the trian pipeline

### ISSUE THREAD

**Issue by wyh196646** [LongxiZhou/DLPE-method#2]
could you supply the training code? My email is wangyuhao@bupt.edu.cn

**Comment by LongxiZhou** [comment]
Please see the updated files in our Github link https://github.com/LongxiZhou/DLPE-method/blob/master/models/Unet_2D/training_control.py. This is for the training of the blood vessels. 

But you should understand that DLPE contains multiple segmentation models, including lungs, heart, COVID-19 lesions, airways, blood vessels, and sub-visual lesions. And each segmentation model needs different hyper-parameters, loss functions and training protocol. All models are based on 2D U-net, and on the Google drive, there are 27 trained models that together form the DLPE method. This means it is very hard to offer you a pipeline for "one click to train all models".

Luckily, all models for DLPE are robust and not sensitive to hyper-parameters. We offered the detailed guidance for training all models: all tricks and all hyper-parameters are explained: https://static-content.springer.com/esm/art%3A10.1038%2Fs42256-022-00483-7/MediaObjects/42256_2022_483_MOESM1_ESM.pdf

We recommend you to read the guidance first, and use your own code instead. For example, the idea of "feature-enhanced" loss is easy to understand, but our implementation may be hard to understand. If you want, you can see https://github.com/LongxiZhou/DLPE-method/blob/master/sample_manager/loss_weight_cal.py for our implementation of "feature-enhanced loss". You may try other variant of such loss, like during training, change voxel-wise penalty to require the loss caused by non-airway, surface of airway and inside airway to be the same. I can tell you these variant can also reach satisfactory performance.

We hope your research goes smoothly!

Best
Longxi

**Comment by wyh196646** [comment]
really appreciate about your reply
